lib/pure/os.nim

```diff
@@ -3187,38 +3187,56 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
-proc sameFileContent*(path1, path2: string): bool {.rtl, extern: "nos$1",
-  tags: [ReadIOEffect], noWeirdTarget.} =
-  ## Returns true if both pathname arguments refer to files with identical
-  ## binary content.
+proc sameFileContent*(path1, path2: string; checkSize = false; checkFiles = false; bufferSize = 8192.Positive): bool {.
```

Comment:
this cancels out what I did here https://github.com/nim-lang/Nim/pull/16023 ;instead:
```nim
proc sameFileContent*(path1, path2: string; checkSize = false; checkFiles = false; bufferSize = -1): bool
  var bufferSize = bufferSize
  ...
  if bufferSize == -1:
    bufSize = getFileInfo(a).blockSize
```